Feature/game logic

kbjude opened the pull request:
- The implementation of the game logic/ game Engine
- A few changes to the main file to relate with the game logic

me-skit committed:
Add the Game class inside a game.rb file

This commit adds a new class to the project, the Game class inside a new file,
the game.rb file, for the Game class we have created a #initialize, #free?,

me-skit committed:
Add #winer? method and improved #move in Game class

In this commit, I added #winer? a method, improved the #move one and made

me-skit committed:
Add the #board method in the Game class

In this commit, I added a public method called #board wich can be used to draw
the board, in addition, I changed the name of the #winer? method with #winner?.

kbjude committed:
Merge remote-tracking branch 'origin' into feature/game_logic

kbjude committed:
implementing the game Engine

kbjude committed:
fixing linting errors

me-skit committed:
Add missing directive in the main executable file

This commit adds the '#!/user/bin/env ruby' directive at the begining of the
main executable, in addition, I created a file for the Player class wich was
inside the main file and deleted the .gitkeep inside the lib folder.

me-skit committed:
Fix the commit before this

The commit before this the changes were not stored in the remote repo because
I was in a subfolder when I execute the git commands. This commit fix that
mistake.

kobiyoyo commented on lib/player.rb:
```diff
@@ -0,0 +1,11 @@
+class Player
+  attr_reader :character
+  attr_reader :name
+  attr_accessor :score
```
Use attr_readers/attr_writers instead of attr_accessors

meme-es replied:
Could you tell us why to help us understand?

meme-es replied:
By the way, the draw method is in game.board not in the move(player, num), so the code respect the single responsibility.

kobiyoyo replied:
Using it  allow writing that property from outside the class. That variable can be set to anything

meme-es replied:
That is the purpose, I understand the attribute accessor... I mean why 'use attr_reader/attr_writer instead of attr_accessor?'

kbjude replied:
Kobiyoyo do you mean that we work on attr_accessor :score as attr_reader ? And if yes, could you advise according to the Manuel's question

kobiyoyo reviewed (changes requested):
## Hi Team    ![](https://user-images.githubusercontent.com/38801423/62908122-703d4480-bd44-11e9-9115-42025ab7f341.png) 
#### Great Job so far, but there are a few more things required. 
### General
- [x] Lets fix Stickler errors
- [x] Note there should be a winner when the:
-  three horizontal elements  are equal
-  three vertical elements are equal 
- three equal elements form a diagonal, from left and right to right
- [x] This should all display the winner but am been asked to input.(below)
![3](https://user-images.githubusercontent.com/23278161/78828801-2ee95b80-79dd-11ea-9571-f7f7fb6a80ee.png)
![2](https://user-images.githubusercontent.com/23278161/78828805-301a8880-79dd-11ea-84d3-47331deee45d.png)
![1](https://user-images.githubusercontent.com/23278161/78828807-30b31f00-79dd-11ea-88e3-308ceeb7fec0.png)
- [x] All classes are kept in `lib` directory. You have `UserInterface` class in main
- [x]   The code should respect the single responsibility rule for methods. Draw should have its own function,it should  not be  in ` move(player, num)`
Read comments below for more details
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

me-skit committed:
Fix game logic and add a userinterface.rb file

This commit fixes an issue in the logic game after a match of five movements,
in addition, I put the class UserInterface inside its own file in the lib
folder.

meme-es commented:
### Lint error

The lint error shows the next info, could a reviewer help us to know how to solve it? Thanks in advance!

![Selection_005](https://user-images.githubusercontent.com/32546183/78834978-9f2bb900-79ac-11ea-90f1-837b03c0aa1b.png)

Torres-ssf reviewed (approved):
### Project Approved 👍 
Hi there team ✋ 
No job implementing the requested changes. I tested your game and everything seems to work. 
- About the attr_acessor, I think for score it makes sense to be used since it needs to be both read and updated.
- About the linter errors: Rubocop added some new cops and you need to explicitly tell if they are going to be enabled or not. For now, we can list them as disabled.
Ex:
```
Lint/RaiseException:
  Enabled: false
```
Do this for all 5 and the linter will stop complaining. I checked you project locally and there aren't any other offences besides these ones, so **I'm approving this project**. But please, before merging, add these new rules to .rubocop.yml file and make a commit in order to make the linter stop complaining.

Good luck with the next milestone.
Happy coding 💻 

**Sergio Torres**

me-skit committed:
Fix lint errors

This commit configure the next cops lints:
- Lint/RaiseException
- Lint/StructureNewOverride
- Style/HashEachMethods
- Style/HashTransformKeys
- Style/HashTransformValues

meme-es commented:
Thanks for the help @Torres-ssf
